### ISSUE TITLE
new BigIntType("string") not working with 6.0.4

### ISSUE THREAD

**Issue by filenwind** [mikro-orm/mikro-orm#5146]
### Describe the bug

I'm pretty sure using same code between 6.0.1 and 6.0.4

when I set: `type: new BigIntType("string")`

6.0.1 will return `string` for me, but 6.0.4 will return `bigint`

I trace the code, found it should happened at here

https://github.com/mikro-orm/mikro-orm/commit/312f293cb9a23115da440c64e0a8ff7e21ae13c2#diff-61ad57ac91248d014b0da6a0a39b4bf5732eab2b0ae92eb7c92206d057c39975R1306-R1308

```js
if (prop.customType instanceof BigIntType && ['string', 'bigint', 'number'].includes(prop.runtimeType.toLowerCase())) {
  prop.customType.mode = prop.runtimeType.toLowerCase() as 'string';
}
```

I try to log like below
```js
if (prop.customType instanceof types_1.BigIntType && ['string', 'bigint', 'number'].includes(prop.runtimeType.toLowerCase())) {
  if (prop.customType.mode.toLowerCase() !== prop.runtimeType.toLowerCase()) {
    console.log(`prop`, prop);
    console.log(`prop.customType.mode`, prop.customType.mode);
    console.log(`prop.runtimeType`, prop.runtimeType);
  }
  prop.customType.mode = prop.runtimeType.toLowerCase();
}
```

and got info
```js
prop {
  name: 'id',
  kind: 'scalar',
  primary: true,
  type: 'BigIntType',
  unsigned: true,
  autoincrement: true,
  runtimeType: 'bigint',
  fieldNames: [ 'id' ],
  customType: BigIntType { mode: 'string' },
  columnTypes: [ 'bigint' ],
  hasConvertToJSValueSQL: false,
  hasConvertToDatabaseValueSQL: false
}
prop.customType.mode string
prop.runtimeType bigint
```

It change by the runtimeType, I don't know how the runtimeType come from

below is my settings
```js
export const TestEntitySchema = new EntitySchema({
  class: TestEntity,
  tableName: "test",
  properties: {
    id: { primary: true, type: new BigIntType("string") },
    ...etc
  },
});
```

If this is intended, how can I avoid that?

Thank you very much!

Btw, the driver is MYSQL~!

### Reproduction

I think the commit history is enough

### What driver are you using?

None

### MikroORM version

6.0.4

### Node.js version

v20.8.1

### Operating system

macOS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by filenwind** [comment]
Sorry about that my eyes broken...but I'm really facing the issue :'(...I will keep looking reason

sorry again...Orz

**Comment by B4nan** [comment]
> I think the commit history is enough

It's the very opposite, I need reproduction, that's always the first step. If you don't provide one, I always need to build it first myself, only then I can try to fix it. Once you have that, you can start debugging what's happening, but before it, it's not adding much value.

**Comment by filenwind** [comment]
Really sorry, here is the reproduce repo

https://github.com/filenwind/mikro-biginttype-not-work

Thanks for your help!

**Comment by filenwind** [comment]
> I see you are using `EntitySchema`, so that on its own could be the problem, I'll try that first. Also provide at least your ORM config and tsconfig, those things matter when it comes to metadata a lot.

Yup, because when I use `Class with decorator` to define Entity, It will very slow at start up (mikorm.init)
I take a long time try to figure out why, but can't find the answer, and thanks god I found `EntitySchema` help! :D

I have more then two hundred of entities, with `Class with decorator` ...I can't remember, maybe will spend 30 sec at start up, but with `EntitySchema` less than 1 sec

I'm using mikro with NEXTJS, using `next build`, `next start` maybe is the clue.

**Comment by B4nan** [comment]
Decorators are fast (it's the same as `EntitySchema`), sounds like you were trying out ts-morph, which is indeed slow on startup as it uses TS compiler API.
